File: app/src/main/java/com/tregula/graedukacyjna/domain/database/dao/CountryDao.java
package com.tregula.graedukacyjna.domain.database.dao;

import com.tregula.graedukacyjna.domain.database.entity.Country;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import io.reactivex.Single;

@Dao
public interface CountryDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(List<Country> countries);

    @Query("SELECT * FROM countries WHERE name LIKE :name")
    Single<Country> loadByName(String name);

    @Query("SELECT * FROM countries ORDER BY name ASC")
    Single<List<Country>> getCountries();
}

File: app/src/main/java/com/tregula/graedukacyjna/domain/database/dao/CountryDao.kt
package com.tregula.graedukacyjna.domain.database.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.tregula.graedukacyjna.domain.database.entity.Country
import io.reactivex.Single

@Dao
interface CountryDao {

    @Query("SELECT * FROM countries ORDER BY name ASC")
    fun getCountries(): Single<List<Country>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(countries: List<Country>)

    @Query("SELECT * FROM countries WHERE name LIKE :name")
    fun loadByName(name: String): Single<Country>
}
